app/shared_channel.go

```diff
@@ -98,6 +99,20 @@ func (a *App) HasRemote(channelID string, remoteID string) (bool, error) {
 	return a.Srv().Store.SharedChannel().HasRemote(channelID, remoteID)
 }
 
+func (a *App) GetRemoteClusterForUser(remoteID string, userID string) (*model.RemoteCluster, *model.AppError) {
+	rc, err := a.Srv().Store.SharedChannel().GetRemoteForUser(remoteID, userID)
+	if err != nil {
+		var nfErr *store.ErrNotFound
+		switch {
+		case errors.As(err, &nfErr):
+			return nil, model.NewAppError("GetRemoteClusterForUser", "api.context.remote_id_invalid.app_error", nil, nfErr.Error(), http.StatusNotFound)
+		default:
+			return nil, model.NewAppError("GetRemoteClusterForUser", "api.context.remote_id_invalid.app_error", nil, err.Error(), http.StatusNotFound)
```

Comment:
Is this still a "not found" error? If so, there is no need for the `switch` block.

Reply:
I understand a single case + default switch might look strange, but that is the defacto standard for our app.  That exact switch occurs in 70 places in the app, including new code.

Reply:
Check latest code, I believe the second case does return internal error.